grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -318,24 +317,20 @@ private void shutdownLbRpc() {
     }
   }
 
-  private void startLbComm(LbAddressGroup lbAddressGroup) {
-    checkNotNull(lbAddressGroup, "lbAddressGroup");
+  private void startLbComm(List<EquivalentAddressGroup> overrideAuthorityEags) {
+    checkNotNull(overrideAuthorityEags, "overrideAuthorityEags");
+    assert !overrideAuthorityEags.isEmpty();
+    String arbitraryAuthority = overrideAuthorityEags.get(0).getAttributes()
```

Comment:
It seems like there's a missing usage of EAG.ATTR_AUTHORITY_OVERRIDE in the overrideAuthorityEags, so that this arbitraryAuthority is actually choosing the authority for all connections. We need to add the missing ATTR_AUTHORITY_OVERRIDE, but it seems we should also modify this authority such that it can't be accidentally load-bearing. How about we append "-notIntendedToBeUsed" to the arbitrary authority, or something similar? (Using the LB_ADDR_AUTHORITY as a base does have the advantage that we can better tell the source of the authority string; so while it'd work fine to completely replace the string with "not-used" that may make some logs less helpful.)

Reply:
Oh no, those constants are camouflaging! Yea sure.